File: implementations/php/src/Data/CaseStudy.php
<?php

declare(strict_types=1);

namespace GraphQL\Compatibility\Data;

use GraphQL\Utils\Utils;

class CaseStudy
{
    public string $caseNumber;

    public string $description;

    public function __construct(array $data)
    {
        Utils::assign($this, $data);
    }
}

File: implementations/hotchocolate/Types/CaseStudy.cs
namespace Products;

public class CaseStudy
{

    public CaseStudy(string caseNumber, string? description)
    {
        CaseNumber = caseNumber;
        Description = description;
    }

    [ID]
    public string CaseNumber { get; }
    public string? Description { get; }

}
